src/spawn_sync.cc

```diff
@@ -954,14 +955,15 @@ int SyncProcessRunner::CopyJsStringArray(Local<Value> js_value,
   if (!js_value->IsArray())
     return UV_EINVAL;
 
-  js_array = js_value.As<Array>()->Clone().As<Array>();
-  length = js_array->Length();
+  input_array = js_value.As<Array>();
+  length = input_array->Length();
+  js_array = Array::New(env()->isolate(), length);
 
   // Convert all array elements to string. Modify the js object itself if
   // needed - it's okay since we cloned the original object.
   for (uint32_t i = 0; i < length; i++) {
-    if (!js_array->Get(i)->IsString())
-      js_array->Set(i, js_array->Get(i)->ToString(env()->isolate()));
+    auto item = input_array->Get(i)->ToString(env()->isolate());
```

Comment:
may not be a bad idea to check the `Get` and `ToString` (`ToLocalChecked` is ugly but may be a good thing to have if you don't want to throw exceptions back to js)